baseapp/baseapp.go

```diff
@@ -842,6 +842,12 @@ func (app *BaseApp) runTx(mode execMode, txBytes []byte) (gInfo sdk.GasInfo, res
 	// fails. Hence, it's execution is deferred.
 	consumeBlockGas := func() {
 		if !blockGasConsumed {
+			defer func() {
+				// log and ignore, see: https://github.com/cosmos/cosmos-sdk/issues/19317
+				if err := recover(); err != nil {
+					ctx.Logger().Error("block gas limit exceeded", "err", err)
+				}
+			}()
 			blockGasConsumed = true
 			ctx.BlockGasMeter().ConsumeGas(
```

Comment:
Sorry, can you please explain again how this could panic, thus making recovery needed?

Reply:
It's related to unused gas refunding, if we include tx by user provided gas limit, they can fill block space with unused gas, the actually used gas is always lower than the block gas limit, we did some hack to include more txs, so we got exceeding block gas limit situation occasionally.
If we plan to remove block gas meter at all, maybe we can simply live with the situation.